File: quickfix/service_center/report/technician_performance_report_monthly/technician_performance_report_monthly.js
// Copyright (c) 2026, Namitha and contributors
// For license information, please see license.txt

// frappe.query_reports["Technician Performance Report Monthly"] = {
// 	filters: [
// 		// {
// 		// 	"fieldname": "my_filter",
// 		// 	"label": __("My Filter"),
// 		// 	"fieldtype": "Data",
// 		// 	"reqd": 1,
// 		// },
// 	],
// };


frappe.query_reports["Technician Performance Report Monthly"] = {
    "filters": [
        {
            "fieldname": "from_date",
            "label": "From Date",
            "fieldtype": "Date",
            "default": frappe.datetime.month_start(),
            "reqd": 1
        },
        {
            "fieldname": "to_date",
            "label": "To Date",
            "fieldtype": "Date",
            "default": frappe.datetime.month_end(),
            "reqd": 1
        },
        {
            "fieldname": "technician",
            "label": "Technician",
            "fieldtype": "Link",
            "options": "Technician"
        }
    ]
};

File: quickfix/service_center/report/technician_performance_report_monthly/technician_performance_report_monthly.py
import frappe

def execute(filters=None):

    columns = [
        {"label": "Technician", "fieldname": "technician", "fieldtype": "Data", "width": 200},
        {"label": "Jobs Completed", "fieldname": "jobs", "fieldtype": "Int", "width": 150},
        {"label": "Total Estimated Cost", "fieldname": "total_cost", "fieldtype": "Currency", "width": 150}
    ]

    conditions = ""

    if filters.get("technician"):
        conditions += " AND assigned_technician = %(technician)s"

    data = frappe.db.sql("""
        SELECT
            assigned_technician as technician,
            COUNT(name) as jobs,
            SUM(estimated_cost) as total_cost
        FROM
            `tabJob Card`
        WHERE
            creation BETWEEN %(from_date)s AND %(to_date)s
            AND status = 'Completed'
            {conditions}
        GROUP BY assigned_technician
    """.format(conditions=conditions), filters, as_dict=1)

    return columns, data